packages/block-library/src/heading/heading-checker.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { speak } from '@wordpress/a11y';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { computeOutlineHeadings } from '../../../editor/src/components/document-outline/index.js';
```

Comment:
This won't work when published to npm. You can't reference private methods from another package this way. I guess this method should be copied over here instead as the first iteration.

Reply:
Solve it. But I do not think it is a good coding style to copy-paste the existing function. Is it possible to just export the function and import it?

Reply:
> But I do not think it is a good coding style to copy-paste the existing function.

Yes, 100% correct.

> Is it possible to just export the function and import it?

There is an alternative, see my other comment: https://github.com/WordPress/gutenberg/pull/14889#discussion_r274377528